voc/library.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 Fraunhofer AISEC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//           $$\                           $$\ $$\   $$\
+//           $$ |                          $$ |\__|  $$ |
+//  $$$$$$$\ $$ | $$$$$$\  $$\   $$\  $$$$$$$ |$$\ $$$$$$\    $$$$$$\   $$$$$$\
+// $$  _____|$$ |$$  __$$\ $$ |  $$ |$$  __$$ |$$ |\_$$  _|  $$  __$$\ $$  __$$\
+// $$ /      $$ |$$ /  $$ |$$ |  $$ |$$ /  $$ |$$ |  $$ |    $$ /  $$ |$$ | \__|
+// $$ |      $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$\ $$ |  $$ |$$ |
+// \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+//  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+//
+// This file is part of Clouditor Community Edition.
+
+package voc
+
+// Library is an entity in our Cloud ontology. It encapsulates the (source) code of a library, similar to an application.
+// TODO(oxisto): Add this to the ontology and auto-generate it
+type Library struct {
+	*Resource
+	Functionalities     []*Functionality `json:"functionalities"`
+	ProgrammingLanguage string           `json:"programmingLanguage"`
+	TranslationUnits    []ResourceID     `json:"translationUnits"`
+	Dependencies        []ResourceID     `json:"dependencies"`
+}
```

Comment:
I would prefer some small comments to clarify what they mean, e.g. for `Functionalities` and `TranslationUnits`. But, on the other hand, they would probably be overwritten by the auto generation anyway.

Reply:
Yes. I think we had a beta branch of the auto-generator that generates also comments, but development there has stalled for a while.